x-pack/plugins/security_solution/server/lib/detection_engine/rule_actions/legacy_get_bulk_rule_actions_saved_object.ts

```diff
@@ -31,21 +31,30 @@ export const legacyGetBulkRuleActionsSavedObject = async ({
   alertIds,
   savedObjectsClient,
 }: LegacyGetBulkRuleActionsSavedObject): Promise<Record<string, LegacyRulesActionsSavedObject>> => {
-  const filter = buildChunkedOrFilter(
-    `${legacyRuleActionsSavedObjectType}.attributes.ruleAlertId`,
-    alertIds
-  );
+  const references = alertIds.map<SavedObjectsFindOptionsReference>((alertId) => ({
+    id: alertId,
+    type: 'alert',
+  }));
   const {
     // eslint-disable-next-line @typescript-eslint/naming-convention
     saved_objects,
   } = await savedObjectsClient.find<LegacyIRuleActionsAttributesSavedObjectAttributes>({
     type: legacyRuleActionsSavedObjectType,
     perPage: 10000,
-    filter,
+    hasReference: references,
   });
   return saved_objects.reduce(
     (acc: { [key: string]: LegacyRulesActionsSavedObject }, savedObject) => {
-      acc[savedObject.attributes.ruleAlertId] = legacyGetRuleActionsFromSavedObject(savedObject);
+      const ruleAlertId = savedObject.references.find((reference) => {
+        // Find the first rule alert and assume that is the one we want since we should only ever have 1.
+        return reference.type === 'alert';
```

Comment:
Yeah, I can. You are correct that we can't guarantee that there aren't more than 1. I figured if there was more than 1 I would just pull 1. If there's 2 or more, I think we have to just pull the first 1. Best I can do would be to check if there is more than 1 and then write out an error message in case we begin encountering it we will see it fairly quickly in the log files.